pkg/noderesourcetopology/filter.go

```diff
@@ -0,0 +1,263 @@
+		policies = append(policies, apiconfig.TopologyManagerPolicy(policy))
+	}
+	return policies
+}
+
+func extractResources(zone topologyv1alpha1.Zone) v1.ResourceList {
+	res := make(v1.ResourceList)
+	for _, resInfo := range zone.Resources {
+		quantity, err := resource.ParseQuantity(resInfo.Allocatable.String())
+		if err != nil {
+			klog.Errorf("Failed to parse %s", resInfo.Allocatable.String())
+			continue
+		}
+		res[v1.ResourceName(resInfo.Name)] = quantity
+	}
+	return res
+}
+
+func SingleNUMAContainerLevelHandler(pod *v1.Pod, zones topologyv1alpha1.ZoneList) *framework.Status {
+	klog.V(5).Infof("Single NUMA node handler")
+	containers := []v1.Container{}
+	containers = append(pod.Spec.InitContainers, pod.Spec.Containers...)
+
+	// prepare NUMANodes list from zoneMap
+	nodes := createNUMANodeList(zones)
+	qos := v1qos.GetPodQOS(pod)
+	for _, container := range containers {
+		bitmask := bm.NewEmptyBitMask()
+		bitmask.Fill()
+
+		checkResourcesForNUMANodes(bitmask, nodes, container.Resources.Requests, qos)
+		if bitmask.IsEmpty() {
+			// definitly we can't align container, so we can't align a pod
+			return framework.NewStatus(framework.Unschedulable, fmt.Sprintf("Cannot align container: %s", container.Name))
+		}
+	}
+	return nil
+}
+
+func checkResourcesForNUMANodes(bitmask bm.BitMask, nodes NUMANodeList, resources v1.ResourceList, qos v1.PodQOSClass) {
+	zeroQuantity := resource.MustParse("0")
+	for resource, quantity := range resources {
+		resourceBitmask := bm.NewEmptyBitMask()
+		for _, numaNode := range nodes {
+			numaQuantity, ok := numaNode.Resources[resource]
+			// if can't find requested resource on the node - skip (don't set it as available NUMA node)
+			// if unfound resource has 0 quantity probably this numa node can be considered
+			if !ok && quantity.Cmp(zeroQuantity) != 0 {
+				continue
+			}
+			// Check for the following:
+			// 1. set numa node as possible node if resource is memory or Hugepages (until memory manager will not be merged and
+			// memory will not be provided in CRD
+			// 2. set numa node as possible node if resource is cpu and it's not guaranteed QoS, since cpu will flow
+			// 3. set numa node as possible node if zero quantity for non existing resource was requested (TODO check topology manaager behaviour)
+			// 4. otherwise check amount of resources
+			if resource == v1.ResourceMemory ||
+				strings.HasPrefix(string(resource), string(v1.ResourceHugePagesPrefix)) ||
+				resource == v1.ResourceCPU && qos != v1.PodQOSGuaranteed ||
+				quantity.Cmp(zeroQuantity) == 0 ||
+				numaQuantity.Cmp(quantity) >= 0 {
+				resourceBitmask.Add(numaNode.NUMAID)
+			}
+		}
+		bitmask.And(resourceBitmask)
+	}
+}
+
+func SingleNUMAPodLevelHandler(pod *v1.Pod, zones topologyv1alpha1.ZoneList) *framework.Status {
+	klog.V(5).Infof("Pod Level Resource handler")
+	containers := []v1.Container{}
+	containers = append(pod.Spec.InitContainers, pod.Spec.Containers...)
```

Comment:
Ditto, you can iterate `append(pod.Spec.InitContainers, pod.Spec.Containers...)` directly in the for loop below.